### DISCUSSION TITLE
Duplicate pageContext

### DISCUSSION THREAD

**Discussion by marek-hanzal** [vikejs/vike#1696] [Discussion]
This one is strange:
I've found in docs I should **not** use `onBeforePrerenderStart` for returning `pageContext` even it would be quite cool do the same stuff in one run, so I'm using `+data`.

...buuut.....

It get's generated twice: once as json file with `pageContext`, secondly embedded directly in `html` file. As I'm sending static translations, this is quite ugly as for each page I would get tons of duplicate data (I get why it's generated per page, but `pageContext` + html is out of my mind :O ).

![image](https://github.com/vikejs/vike/assets/13855126/1a4ab0dd-6d55-4711-9450-6dde70fc6afd)

Is there any reason for this behavior?

Thank you :)

**Comment by brillout** [comment]
In an SSR context it makes sense to embed the pageContext data into the HTML (saving a round trip), but indeed for pre-rendered pages it would make more sense to avoid this. It‘s on the radar although not the a priority at the moment.

**Reply by marek-hanzal** [reply]
That's ok, not a big deal for me, but thank you for reply :).
